File: android/sdk/src/main/java/com/ble/notification/crypto/KeyDerivation.kt
package com.ble.notification.crypto

/**
 * Derives per-package AES keys from a BLE notification sync salt using HKDF-SHA256.
 */
object KeyDerivation {

    /**
     * Derive a 32-byte AES key for the given package name.
     *
     * Uses the project-wide [NativeCrypto.SALT] and HKDF-SHA256 to produce a
     * deterministic, domain-separated key per package.
     *
     * @param packageName application package name (e.g. "com.example.app")
     * @return 32-byte derived key
     * @throws IllegalStateException if native HKDF fails
     */
    fun deriveKey(packageName: String): ByteArray {
        return NativeCrypto.deriveKey(packageName)
    }
}

File: windows/BleNotificationWin/Crypto/KeyDerivation.cs
using System;

namespace BleNotificationWin.Crypto;

/// <summary>
/// Derives per-package AES keys from a BLE notification sync salt using HKDF-SHA256.
/// </summary>
public static class KeyDerivation
{
    /// <summary>
    /// Derive a 32-byte AES key for the given package name.
    /// Uses the project-wide salt and HKDF-SHA256 to produce a deterministic,
    /// domain-separated key per package.
    /// </summary>
    /// <param name="packageName">Application package name (e.g. "com.example.app")</param>
    /// <returns>32-byte derived key</returns>
    /// <exception cref="InvalidOperationException">Thrown if native HKDF fails</exception>
    public static byte[] DeriveKey(string packageName)
    {
        return LibTomCrypt.DeriveKey(packageName);
    }
}
